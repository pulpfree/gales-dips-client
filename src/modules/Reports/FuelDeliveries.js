import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import ReportSelectors from './ReportSelectors'
import { styles as ms } from '../../styles/main'


class FuelDeliveries extends Component {

  renderReport(pathname) {
    const date = pathname.split('/')[3]
    const storeID = pathname.split('/')[4]

    if (!date || ! storeID) return

    return (
      <div>
        <Typography variant="body1">date: {date}</Typography>
        <Typography variant="body1">storeID: {storeID}</Typography>
      </div>
    )
  }

  render() {

    const { pathname } = this.props.location

    return (
      <div>
        <Typography
            gutterBottom
            variant="subheading"
        >Fuel Deliveries</Typography>
        <ReportSelectors />
        {this.renderReport(pathname)}
      </div>
    )
  }
}

FuelDeliveries.propTypes = {
  classes:  PropTypes.object.isRequired,
  location: PropTypes.object,
}

export default withStyles(ms)(FuelDeliveries)
