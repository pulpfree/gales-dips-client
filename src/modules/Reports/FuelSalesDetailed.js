import React, { Component } from 'react'
import PropTypes from 'prop-types'

import gql from 'graphql-tag'
import moment from 'moment'
import { graphql } from 'react-apollo'

import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import * as utils from '../../utils/utils'
import Loader from '../Common/Loader'
import ReportSelectors from './ReportSelectors'
import { FUEL_TYPE_LIST } from '../../config/constants'


const FSD_REPORT_QUERY = gql`
query FuelSaleDetailedReport($date: String!, $stationID: String!) {
  fuelSaleDetailedReport(date: $date, stationID: $stationID) {
    stationID
    fuelTypes
    weekSales {
      sales {
        date
        sales
      }
      totals
      yearWeek
    }
  }
}
`

const Report = ({ data, classes }) => {
  if (!data) return null
  if (data && data.loading) { return <div className={classes.container}><Loader /></div> }

  if (!data.fuelSaleDetailedReport) {
    return (
      <div className={classes.container}>
        <Typography
          variant="body2"
        >There is no data available for the specified date.
        </Typography>
      </div>
    )
  }

  const fts = utils.setOrderedFuelTypes(data.fuelSaleDetailedReport.fuelTypes, FUEL_TYPE_LIST)
  return (
    <div className={classes.container}>
      <Typography
        gutterBottom
        variant="h5"
      >Fuel Sales Station Detail
      </Typography>
      {data.fuelSaleDetailedReport.weekSales.map((sale, i) => (
        <Week
          classes={classes}
          data={sale}
          fuelTypes={fts}
          key={i}
        />
      ))}
    </div>
  )
}
Report.propTypes = {
  classes: PropTypes.object.isRequired,
  data: PropTypes.object,
}

const WeekHeader = ({ wkNo, fuelTypes, classes }) => (
  <div className={classes.weekHdrRow}>
    <div className={classes.weekHdrCell}>Week {wkNo}</div>
    {fuelTypes.map(ft => (
      <div
        className={classes.weekDataRt}
        key={ft}
      >{ft}
      </div>
    ))}
  </div>
)
WeekHeader.propTypes = {
  classes: PropTypes.object.isRequired,
  fuelTypes: PropTypes.array.isRequired,
  wkNo: PropTypes.string.isRequired,
}

const Week = ({ data, fuelTypes, classes }) => {
  const wkNo = data.yearWeek.toString().substring(4)
  return (
    <Paper className={classes.weekContainer}>
      <WeekHeader
        classes={classes}
        fuelTypes={fuelTypes}
        wkNo={wkNo}
      />
      {data.sales.map(sale => (
        <div
          className={classes.weekDataRow}
          key={sale.date}
        >
          <div className={classes.weekDataCell}>{moment(sale.date.toString()).format('MMM DD')}</div>
          {fuelTypes.map(ft => (
            <div
              className={classes.weekDataRt}
              key={ft}
            >{utils.fmtNumber(sale.sales[ft], 0, true)}
            </div>
          ))}
        </div>
      ))}
      <WeekSummary
        classes={classes}
        data={data.totals}
        fuelTypes={fuelTypes}
      />
    </Paper>
  )
}
Week.propTypes = {
  classes: PropTypes.object.isRequired,
  data: PropTypes.object.isRequired,
  fuelTypes: PropTypes.array.isRequired,
}

const WeekSummary = ({ data, fuelTypes, classes }) => (
  <div className={classes.weekSummary}>
    <div className={classes.weekDataCell} >{'  '}</div>
    {fuelTypes.map(ft => (
      <div
        className={classes.weekDataRt}
        key={ft}
      >{utils.fmtNumber(data[ft], 0, true)}
      </div>
    ))}
  </div>
)
WeekSummary.propTypes = {
  classes: PropTypes.object.isRequired,
  data: PropTypes.object.isRequired,
  fuelTypes: PropTypes.array.isRequired,
}


class FuelSalesDetailed extends Component {
  render() {
    const { classes, data } = this.props

    if (data && data.error) {
      return <p>Data Error :(</p>
    }

    return (
      <div className={classes.mainContainer}>
        <ReportSelectors />
        <Report
          classes={classes}
          data={data}
        />
      </div>
    )
  }
}

FuelSalesDetailed.propTypes = {
  classes: PropTypes.object.isRequired,
  data: PropTypes.object,
  history: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
}

const styles = theme => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    fontFamily: theme.typography.fontFamily,
    marginTop: theme.spacing.unit * 5,
    width: theme.spacing.unit * 55,
  },
  mainContainer: {
    width: 600,
    margin: 'auto',
  },
  weekContainer: {
    display: 'flex-inline',
    marginBottom: theme.spacing.unit * 5,
    paddingBottom: theme.spacing.unit * 2,
    paddingTop: theme.spacing.unit * 2,
  },
  weekHdrRow: {
    borderBottom: 'solid 1px',
    borderBottomColor: theme.palette.grey['300'],
    color: theme.palette.grey['700'],
    display: 'flex',
    flexDirection: 'row',
    fontWeight: 400,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: theme.spacing.unit * 2,
  },
  weekHdrCell: {
    display: 'flex',
    flex: 1,
    textAlign: 'right',
  },
  weekDataRow: {
    borderBottom: 'solid 1px',
    borderBottomColor: theme.palette.grey['300'],
    display: 'flex',
    flexDirection: 'row',
    paddingBottom: theme.spacing.unit,
    paddingTop: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: theme.spacing.unit * 2,
    '&:hover': {
      backgroundColor: theme.palette.grey['100'],
    },
  },
  weekDataCell: {
    flex: 1,
  },
  weekDataRt: {
    flex: 1,
    textAlign: 'right',
  },
  weekSummary: {
    display: 'flex',
    flexDirection: 'row',
    fontWeight: 500,
    paddingTop: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: theme.spacing.unit * 2,
  },
})

export default graphql(FSD_REPORT_QUERY, {
  skip: props => props.location.pathname.split('/').length < 5,
  options: (props) => {
    const prts = utils.extractPathParts(props.location.pathname, 3)
    return ({
      variables: {
        date: moment(prts[0]).format('YYYY-MM-DD'),
        stationID: prts[1],
      },
    })
  },
})(withStyles(styles)(FuelSalesDetailed))
