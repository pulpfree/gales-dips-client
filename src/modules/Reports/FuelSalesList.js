import React, { Component } from 'react'
import PropTypes from 'prop-types'

import gql from 'graphql-tag'
import moment from 'moment'
import { graphql } from 'react-apollo'
import { Link } from 'react-router-dom'

import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import Loader from '../Common/Loader'
import ReportSelectors from './ReportSelectors'
import * as utils from '../../utils/utils'


const FSL_REPORT_QUERY = gql`
query FuelSaleListReport($date: String!) {
  fuelSaleListReport(date: $date) {
    periodHeader
    sales {
      fuelPrices
      periods
      stationID
      stationName
      stationTotal
    }
    periodTotals
    totalsByFuel
  }
}
`

const Report = ({ classes, data, date }) => {

  if (!data) return null
  if (data && data.loading)
    return <div className={classes.container}><Loader /></div>

  if (!data.fuelSaleListReport) {
    return (
      <div className={classes.container}>
        <Typography>There is no data available for the specified date.</Typography>
      </div>
    )
  }

  const sales = data.fuelSaleListReport.sales

  return (
    <div className={classes.container}>
      <Typography
          gutterBottom
          variant="h5"
      >Fuel Sales Station List</Typography>
      <Paper className={classes.reportContainer}>
        <Typography
            className={classes.title}
            gutterBottom
            variant="h6"
        >No-Lead Report</Typography>
        <PeriodHdr
            classes={classes}
            data={data.fuelSaleListReport.periodHeader}
        />
        {sales.map((sale, i) => (
          <SalesRow
              classes={classes}
              data={sale}
              date={date}
              key={i}
              type="NL"
          />
        ))}
        <SummaryRow
            classes={classes}
            data={data.fuelSaleListReport}
            type="NL"
        />
      <Typography>*NOTE: Values in brackets are fuel price average for period</Typography>
      </Paper>
      <br />
      <Paper className={classes.reportContainer}>
        <Typography
            className={classes.title}
            gutterBottom
            variant="h6"
        >Diesel Report</Typography>
        <PeriodHdr
            classes={classes}
            data={data.fuelSaleListReport.periodHeader}
        />
        {sales.map((sale, i) => (
          <SalesRow
              classes={classes}
              data={sale}
              date={date}
              key={i}
              type="DSL"
          />
        ))}
        <SummaryRow
            classes={classes}
            data={data.fuelSaleListReport}
            type="DSL"
        />
      </Paper>
    </div>
  )
}
Report.propTypes = {
  classes:  PropTypes.object.isRequired,
  data:     PropTypes.object,
  date:     PropTypes.string,
}

const PeriodHdr = ({ classes, data }) => {

  let periods = []
  for (const period in data) periods.push(data[period])

  return (
    <div className={classes.periodHdrRow}>
      <div
          className={classes.periodHdrCell}
          style={{flex: .75}}
      />
      {periods.map((period, i) => (
        <div
            className={classes.periodHdrCell}
            key={i}
        >
          {moment(period.startDate).format('MM/DD')} - {moment(period.endDate).format('MM/DD')}
        </div>
      ))}
      <div
          className={classes.dataCell}
          style={{flex: .75}}
      />
    </div>
  )
}
PeriodHdr.propTypes = {
  classes:  PropTypes.object.isRequired,
  data:     PropTypes.object,
}

const SalesRow = ({ classes, data, date, type }) => {

  // Skip stations that don't have diesel when doing diesel report
  if (type === 'DSL' &&  data.stationTotal.DSL <= 0) return null
  const sales = []
  const link = `/reports/fuel-sales-detailed/${date}/${data.stationID}`

  for (const period in data.periods) {
    sales.push({
      amount: data.periods[period].sales[type],
      fuelPrice: type === 'NL' ? data.fuelPrices.prices[period] : null,
    })
  }

  return (
    <div className={classes.dataRow}>
      <div className={classes.dataCellStation}>
        <Link to={link}>{data.stationName}</Link>
      </div>
      {sales.map((sale, i) => (
        <div
            className={classes.dataCell}
            key={i}
        >
          {utils.fmtNumber(sale.amount, 0, true)}
          {sale.fuelPrice && <span className={classes.fuelPrice}>({utils.fmtNumber(sale.fuelPrice, 1)})</span>}
        </div>
      ))}
      <div
          className={classes.dataCell}
          style={{flex: .75}}
      >
        {utils.fmtNumber(data.stationTotal[type], 0 , true)}
      </div>
    </div>
  )
}
SalesRow.propTypes = {
  classes:  PropTypes.object.isRequired,
  data:     PropTypes.object,
  date:     PropTypes.string.isRequired,
  type:     PropTypes.string.isRequired,
}

const SummaryRow = ({ classes, data, type }) => {

  let sums = []
  for (const sum in data.periodTotals) sums.push(data.periodTotals[sum][type])
  const total = data.totalsByFuel[type]

  return (
    <div className={classes.dataSummaryRow}>
      <div className={classes.dataCellStation} />
      {sums.map((sum, i) => (
        <div
            className={classes.dataTotalCell}
            key={i}
        >
          {utils.fmtNumber(sum, 0, true)}
        </div>
      ))}
      <div
          className={classes.dataTotalCell}
          style={{flex: .75}}
      >
        {utils.fmtNumber(total, 0, true)}
      </div>
    </div>
  )
}
SummaryRow.propTypes = {
  classes:  PropTypes.object.isRequired,
  data:     PropTypes.object,
  type:     PropTypes.string.isRequired,
}

class FuelSalesList extends Component {

  render() {

    let dte
    const { classes, data } = this.props
    const prts = utils.extractPathParts(this.props.location.pathname, 3)

    if (prts) {
      dte = prts[0]
    }

    return (
      <div className={classes.mainContainer}>
        <ReportSelectors
            hideStation
        />
        <Report
            classes={classes}
            data={data}
            date={dte}
        />
      </div>
    )
  }
}

FuelSalesList.propTypes = {
  classes:  PropTypes.object.isRequired,
  data:     PropTypes.object,
  history:  PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  match:    PropTypes.object.isRequired,
}

const styles =  theme => ({
  container: {
    display:        'flex',
    flexDirection:  'column',
    fontFamily:     theme.typography.fontFamily,
    marginTop:      theme.spacing.unit * 3,
    width:          theme.spacing.unit * 55,
  },
  mainContainer: {
    width:  900,
    margin: 'auto',
  },
  dataRow: {
    borderBottom:       'solid 1px',
    borderBottomColor:  theme.palette.grey['300'],
    display:            'flex',
    flexDirection:      'row',
    paddingBottom:      theme.spacing.unit,
    paddingTop:         theme.spacing.unit,
    paddingLeft:        theme.spacing.unit * 2,
    paddingRight:       theme.spacing.unit * 2,
    '&:hover': {
      backgroundColor: theme.palette.grey['100'],
    },
  },
  dataSummaryRow: {
    display:            'flex',
    flexDirection:      'row',
    paddingBottom:      theme.spacing.unit,
    paddingTop:         theme.spacing.unit,
    paddingLeft:        theme.spacing.unit * 2,
    paddingRight:       theme.spacing.unit * 2,
  },
  dataCell: {
    flex:       1,
    textAlign:  'right',
  },
  dataTotalCell: {
    flex:       1,
    fontWeight: 550,
    textAlign:  'right',
  },
  dataCellStation: {
    flex:     .75,
  },
  fuelPrice: {
    fontSize:     14,
    paddingLeft:  theme.spacing.unit,
  },
  periodHdrRow: {
    borderBottom:       'solid 1px',
    borderBottomColor:  theme.palette.grey['300'],
    color:              theme.palette.grey['700'],
    display:            'flex',
    flexDirection:      'row',
    fontWeight:         400,
    paddingBottom:      theme.spacing.unit,
    paddingTop:         theme.spacing.unit,
    paddingLeft:        theme.spacing.unit * 2,
    paddingRight:       theme.spacing.unit * 2,
  },
  periodHdrCell: {
    flex:         1,
    textAlign:    'right',
    paddingRight: theme.spacing.unit,
  },
  reportContainer: {
    display:        'flex',
    flexDirection:  'column',
    marginBottom:   theme.spacing.unit * 2,
    paddingBottom:  theme.spacing.unit * 2,
    paddingTop:     theme.spacing.unit * 2,
    width:          900,
  },
  title: {
    paddingLeft:        theme.spacing.unit * 2,
  },
})

export default graphql(FSL_REPORT_QUERY, {
  skip: props => props.location.pathname.split('/').length < 4,
  options: (props) => {
    const prts = utils.extractPathParts(props.location.pathname, 3)
    return ({
      variables: {
        date: prts[0],
      },
    })
  },
})(withStyles(styles)(FuelSalesList))
