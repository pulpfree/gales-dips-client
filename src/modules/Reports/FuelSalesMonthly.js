import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import ReportSelectors from './ReportSelectors'
import { styles as ms } from '../../styles/main'

class FuelSalesMonthly extends Component {

  renderReport() {
    const { pathname } = this.props.location
    const date = pathname.split('/')[3]

    if (!date) return

    return (
      <div>
        <Typography variant="body1">date: {date}</Typography>
      </div>
    )
  }

  render() {

    const { history, match } = this.props

    return (
      <div>
        <Typography
            gutterBottom
            variant="subheading"
        >Monthly Fuel Sales</Typography>
        <ReportSelectors
            hideStation
            history={history}
            match={match}
        />
        {this.renderReport()}
      </div>
    )
  }
}

FuelSalesMonthly.propTypes = {
  classes:  PropTypes.object.isRequired,
  history:  PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  match:    PropTypes.object.isRequired,
}

export default withStyles(ms)(FuelSalesMonthly)
