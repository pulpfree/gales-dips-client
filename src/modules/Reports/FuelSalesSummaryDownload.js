import React, { Component } from 'react'
import PropTypes from 'prop-types'

import moment from 'moment'
import gql from 'graphql-tag'
import { graphql } from 'react-apollo'

import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import Save from '@material-ui/icons/Save'


const DOWNLOAD_REPORT = gql`
mutation FuelSaleRangeSummaryDownload($dateFrom: String!, $dateTo: String!) {
fuelSaleRangeSummaryDownload(dateFrom: $dateFrom, dateTo: $dateTo) {
    dateStart
    dateEnd
    reportLink
  }
}
`

const styles = theme => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    fontFamily: theme.typography.fontFamily,
    margin: 'auto',
    width: 600,
  },
})

class FuelSalesDwnld extends Component {
  constructor(props) {
    super(props)

    this.state = {
      // month: moment(new Date()).subtract(1, 'months').format('MM'),
    }
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange = (event) => {
    this.setState(() => ({ [event.target.name]: event.target.value, toasterMsg: '' }))
  }

  handleSubmit = () => {

  }

  render() {
    const { classes } = this.props

    return (
      <div className={classes.container}>
        <Typography
          gutterBottom
          variant="h6"
        >Station Report Download
        </Typography>
      </div>
    )
  }
}
FuelSalesDwnld.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
}

export default withStyles(styles)(FuelSalesDwnld)
