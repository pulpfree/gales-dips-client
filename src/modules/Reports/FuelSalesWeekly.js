import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import ReportSelectors from './ReportSelectors'
import { styles as ms } from '../../styles/main'

class FuelSalesWeekly extends Component {

  renderReport(pathname) {
    const date = pathname.split('/')[3]
    const storeID = pathname.split('/')[4]

    if (!date || ! storeID) return

    return (
      <div>
        <Typography variant="body1">date: {date}</Typography>
        <Typography variant="body1">storeID: {storeID}</Typography>
      </div>
    )
  }

  render() {

    const { pathname } = this.props.location

    return (
      <div>
        <Typography
            gutterBottom
            variant="subheading"
        >Weekly Fuel Sales</Typography>
        <ReportSelectors />
        {this.renderReport(pathname)}
      </div>
    )
  }
}

FuelSalesWeekly.propTypes = {
  classes:  PropTypes.object.isRequired,
  history:  PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  match:    PropTypes.object.isRequired,
}

export default withStyles(ms)(FuelSalesWeekly)