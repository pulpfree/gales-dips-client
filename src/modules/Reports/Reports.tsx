export function Reports() {
  return (
    <div>
      <h2>Reports</h2>
    </div>
  )
}