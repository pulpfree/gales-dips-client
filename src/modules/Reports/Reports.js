import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Route } from 'react-router'

import Divider from '@material-ui/core/Divider'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import FuelDeliveries from './FuelDeliveries'
import FuelSalesList from './FuelSalesList'
import FuelSalesDetailed from './FuelSalesDetailed'
import Header from '../Header/Header'
import OverShortMonthly from './OverShortMonthly'
import OverShortAnnually from './OverShortAnnually'
import PropaneSalesMonthly from './PropaneSalesMonthly'
import PropaneSalesAnnual from './PropaneSalesAnnual'
import ReportMenu from './ReportMenu'
import StationReportDownload from './StationReportDownload'
import PropaneReportDownload from './PropaneReportDownload'

import { styles as ms } from '../../styles/main'


class Reports extends Component {

  render() {

    const { classes, match } = this.props

    return (
      <div style={{display: 'flex', flexDirection: 'column'}}>
        <Header />
        <Paper className={classes.paper}>
          <Typography
              gutterBottom
              variant="h5"
          >Reports</Typography>
          <ReportMenu />
          <Divider /><br />
          <div>
            <Route
                component={FuelSalesDetailed}
                path={`${match.url}/fuel-sales-detailed`}
            />
            <Route
                component={FuelSalesList}
                path={`${match.url}/fuel-sales-list`}
            />
            <Route
                component={OverShortMonthly}
                path={`${match.url}/overshort-monthly`}
            />
            <Route
                component={OverShortAnnually}
                path={`${match.url}/overshort-annually`}
            />
            <Route
                component={FuelDeliveries}
                path={`${match.url}/fuel-deliveries`}
            />
            <Route
                component={PropaneSalesAnnual}
                path={`${match.url}/propane-sales-annual`}
            />
            <Route
                component={PropaneSalesMonthly}
                path={`${match.url}/propane-sales-monthly`}
            />
            <Route
                component={StationReportDownload}
                path={`${match.url}/report-download-station`}
            />
            <Route
                component={PropaneReportDownload}
                path={`${match.url}/report-download-propane`}
            />
          </div>
        </Paper>
      </div>
    )
  }
}

Reports.propTypes = {
  classes: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
}

export default withStyles(ms)(Reports)
