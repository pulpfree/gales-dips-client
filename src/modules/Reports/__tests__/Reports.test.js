/* eslint no-undef: "off" */

import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'

import { shallow } from 'enzyme'
import Reports from '../Reports'

describe('Reports', () => {
  let mountedC
  let props
  const reportsC = () => {
    if (!mountedC) {
      mountedC = shallow(
        <Router><Reports {...props} /></Router>
      )
    }
    return mountedC
  }

  beforeEach(() => {
    props = {
      match: {},
      history: {},
    }
    mountedC = undefined
  })

  it('matches snapshot', () => {
    expect(reportsC()).toMatchSnapshot()
  })

})
