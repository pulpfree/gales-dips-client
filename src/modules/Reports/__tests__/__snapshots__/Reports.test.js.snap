// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter>
    <WithStyles(Reports)
        history={Object {}}
        match={Object {}}
    />
</MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Reports)
        history={Object {}}
        match={Object {}}
/>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "d90qsw",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "d90qsw",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "history": Object {},
        "match": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Reports)
          history={Object {}}
          match={Object {}}
/>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "d90qsw",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "d90qsw",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "history": Object {},
          "match": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
